micro-rdk-server/esp32/main.rs

```diff
@@ -39,92 +44,80 @@ mod esp32 {
     }
 
     pub(crate) fn main_esp32() {
-        micro_rdk::esp32::esp_idf_svc::sys::link_patches();
+        esp_idf_svc::sys::link_patches();
+        esp_idf_svc::log::EspLogger::initialize_default();
 
-        micro_rdk::esp32::esp_idf_svc::log::EspLogger::initialize_default();
+        esp_idf_svc::sys::esp!(unsafe {
+            esp_idf_svc::sys::esp_vfs_eventfd_register(
+                &esp_idf_svc::sys::esp_vfs_eventfd_config_t { max_fds: 5 },
+            )
+        })
+        .unwrap();
+
+        #[cfg(feature = "qemu")]
+        let _network = {
+            use micro_rdk::esp32::esp_idf_svc::hal::prelude::Peripherals;
+            log::info!("creating eth object");
+            let sys_loop = esp_idf_svc::eventloop::EspEventLoop::take().unwrap();
+            let eth = EspEth::wrap(
+                EthDriver::new_openeth(Peripherals::take().unwrap().mac, sys_loop.clone()).unwrap(),
+            )
+            .unwrap();
+            eth_configure(&sys_loop, eth).unwrap()
+        };
 
         let mut r = Box::<ComponentRegistry>::default();
         register_examples(&mut r);
         let repr = RobotRepresentation::WithRegistry(r);
 
-        {
-            micro_rdk::esp32::esp_idf_svc::sys::esp!(unsafe {
-                micro_rdk::esp32::esp_idf_svc::sys::esp_vfs_eventfd_register(
-                    &micro_rdk::esp32::esp_idf_svc::sys::esp_vfs_eventfd_config_t { max_fds: 5 },
-                )
-            })
-            .unwrap();
+        let storage = NVSStorage::new("nvs").unwrap();
+
+        // At runtime, if the program does not detect credentials or configs in storage,
+        // it will try to load statically compiled values.
```

Comment:
OK, so this is a meaningful behavior change. It used to be that we would just always use the static ones if we had them and step on NVS. Now, it is the other way: if NVS has values, we prefer them. I think that is right: you might have a build with static configs, but then have adjusted information with the installer tool.

Reply:
@npmenard @gvaradarajan @EshaMaharishi @stevebriskin , after talking more with @acmorrow:

If a device is flashed a) from our published binary or b) build from source repo without wifi credentials, it can access provisioning whenever wifi credentials are not present in NVS, as `PASS` and `SSID` will be `None`.

Devices using builds from the project template (which now requires the presence of wifi credentials), will not be able to access provisioning, even if the NVS cache is completely wiped, as `PASS` and `SSID` will always be `Some()`.

In both scenarios, wifi credentials added post-flashing using the installer will be prioritized.

Reply:
Sgtm. My understanding: If the user doesn't provide appropriate credentials, they may enter provisioning mode which will appear to them as being bricked. They will need to re-flash.